src/loaders/utils.ts

```diff
@@ -0,0 +1,161 @@
+  while (len--) {
+    sumSquared += (arr[len] - mean) ** 2;
+  }
+  const sd = (sumSquared / arr.length) ** 0.5;
+
+  // Median, and quartiles via quickselect: https://en.wikipedia.org/wiki/Quickselect.
+  // Odd number lengths should round down the index.
+  const mid = Math.floor(arr.length / 2);
+  const firstQuartileLocation = Math.floor(arr.length / 4);
+  const thirdQuartileLocation = 3 * Math.floor(arr.length / 4);
+
+  quickselect(arr, mid);
+  const median = arr[mid];
+  quickselect(arr, firstQuartileLocation, 0, mid);
+  const q1 = arr[firstQuartileLocation];
+  quickselect(arr, thirdQuartileLocation, mid, arr.length - 1);
+  const q3 = arr[thirdQuartileLocation];
+
+  // Used for "auto" settings.  This is the best parameter I've found experimentally.
+  // I don't think there is a right answer and this feature is common in Fiji.
+  // Also it's best to use a non-zero array for this.
+  const cutoffArr = arr.filter((i: number) => i >= 1);
+  const cutoffPercentile = 0.0005;
+  const topCutoffLocation = Math.floor(
+    cutoffArr.length * (1 - cutoffPercentile)
+  );
+  const bottomCutoffLocation = Math.floor(cutoffArr.length * cutoffPercentile);
+  quickselect(cutoffArr, topCutoffLocation);
+  quickselect(cutoffArr, bottomCutoffLocation, 0, topCutoffLocation);
+  const autoSliders = [
+    cutoffArr[bottomCutoffLocation],
+    cutoffArr[topCutoffLocation]
+  ];
+  return {
+    mean,
+    sd,
+    q1,
+    q3,
+    median,
+    domain: [min, max],
+    autoSliders
+  };
+}
+
+export function ensureArray<T>(x: T | T[]) {
+  return Array.isArray(x) ? x : [x];
+}
+
+/*
+ * Converts 32-bit integer color representation to RGBA tuple.
+ * Used to serialize colors from OME-XML metadata.
+ *
+ * > console.log(intToRgba(100100));
+ * > // [0, 1, 135, 4]
+ */
+export function intToRgba(int: number) {
+  if (!Number.isInteger(int)) {
+    throw Error('Not an integer.');
+  }
+
+  // Write number to int32 representation (4 bytes).
+  const buffer = new ArrayBuffer(4);
+  const view = new DataView(buffer);
+  view.setInt32(0, int, false); // offset === 0, littleEndian === false
+
+  // Take u8 view and extract number for each byte (1 byte for R/G/B/A).
+  const bytes = new Uint8Array(buffer);
+  return Array.from(bytes) as [number, number, number, number];
+}
+
+/*
+ * Helper method to determine whether pixel data is interleaved or not.
+ * > isInterleaved([1, 24, 24]) === false;
+ * > isInterleaved([1, 24, 24, 3]) === true;
+ */
+export function isInterleaved(shape: number[]) {
+  const lastDimSize = shape[shape.length - 1];
+  return lastDimSize === 3 || lastDimSize === 4;
+}
+
+/*
+ * Creates typed labels from DimensionOrder.
+ * > imgMeta.Pixels.DimensionOrder === 'XYCZT'
+ * > getLabels(imgMeta.Pixels) === ['t', 'z', 'c', 'y', 'x']
+ */
+type Sel<
+  Dim extends string
+> = Dim extends `${infer Z}${infer X}${infer A}${infer B}${infer C}`
+  ? [C, B, A]
+  : 'error';
+export function getLabels(dimOrder: OMEXML[0]['Pixels']['DimensionOrder']) {
```

Comment:
Do we want to have an OME specific thing in `utils`?

Reply:
Right nearly all the loader code is OME-Specfic. I'm open to extending this when we have tangible, non-ome code. But for now the more simple the better IMO

Reply:
Also fairly clear from the function signature that it's for OME-XML metadata. You couldn't try to use this any other way.